aea/crypto/ledger_apis.py

```diff
@@ -110,25 +145,30 @@ def is_tx_settled(self, identifier: str, tx_digest: str, amount: int) -> bool:
         """
         assert identifier in self.apis.keys(), "Unsupported ledger identifier."
         is_successful = False
-        try:
-            api = self.apis[identifier]
-            logger.info("Checking the transaction ...")
-            # tx_status = cast(TxStatus, api.tx.status(tx_digest))
-            tx_status = cast(str, api.tx.status(tx_digest))
-            SUCCESSFUL_TERMINAL_STATES = ('Executed', 'Submitted')
-            if tx_status in SUCCESSFUL_TERMINAL_STATES:
-                # TODO: check the amount of the transaction is correct
-                is_successful = True
-            logger.info("Transaction validated ...")
-        except Exception:
-            logger.warning("An error occurred while attempting to check the transaction.")
+        api = self.apis[identifier]
+
+        if identifier == FETCHAI:
+            try:
+                logger.info("Checking the transaction ...")
+                # tx_status = cast(TxStatus, api.tx.status(tx_digest))
+                tx_status = cast(str, api.tx.status(tx_digest))
+                if tx_status in SUCCESSFUL_TERMINAL_STATES:
+                    # TODO: check the amount of the transaction is correct
+                    is_successful = True
+                logger.info("Transaction validated ...")
+            except Exception:
+                logger.warning("An error occurred while attempting to check the transaction.")
+        elif identifier == ETHEREUM:
+            is_successful = True
```

Comment:
Can you add the code for ethereum here?

Reply:
I will once I solve the problem with the transaction.